tests/foreman/api/test_repositories.py

```diff
@@ -315,9 +320,12 @@ def test_positive_sync_mulitple_large_repos(module_target_sat, module_sca_manife
             reposet=constants.REPOS['kickstart'][name]['reposet'],
             releasever=constants.REPOS['kickstart'][name]['version'],
         )
+        all_repo_ids.append(rh_repo_id)
     rh_repos = module_target_sat.api.Repository(id=rh_repo_id).read()
     rh_products = module_target_sat.api.Product(id=rh_repos.product.id).read()
-    assert len(rh_products.repository) == 4
+    assert len(rh_products.repository) >= 4
+    rh_product_repo_ids = [repo.id for repo in rh_products.repository]
+    assert set(all_repo_ids).issubset(rh_product_repo_ids)
     res = module_target_sat.api.ProductBulkAction().sync(
         data={'ids': [rh_products.id]}, timeout=2000
     )
```

Comment:
This looks strange to me
```
    rh_repos = module_target_sat.api.Repository(id=rh_repo_id).read()
                                                   ^ here we take the single last repo from the loop above
    rh_products = module_target_sat.api.Product(id=rh_repos.product.id).read()
                                                   ^ and we read product for that laset repo
    assert len(rh_products.repository) >= 4
                                       ^ assert that we have more than 4 repos in that product?
    rh_product_repo_ids = [repo.id for repo in rh_products.repository]
    assert set(all_repo_ids).issubset(rh_product_repo_ids)
    res = module_target_sat.api.ProductBulkAction().sync(
        data={'ids': [rh_products.id]}, timeout=2000
    )
```
I would need to try it locally to get better understanding and maybe rewrite it.

Reply:
- all of these lines are dependant, as I can see in debug data last repository id i.e `rh_repo_id` used to get product detail and using product details again fetching repositories.
- either need to implement cli way to list all repositories [here](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/cli/repository.py) or keep as it is.

Reply:
So could it be renamed to `rh_repo` and `rh_product` (not plural) instead?
But I still don't get why the product has more than 4 repos in it.

Reply:
ok I will rename it. and there are 4 rhel_8 repos as we are defining/enabling at the begining of this test
```
repo_names = ['rhel8_bos', 'rhel8_aps']
kickstart_names = ['rhel8_bos', 'rhel8_aps']
```

Reply:
when I checked it locally by running whole test module, then the length returns to 5 (rhel9_bos Kickstart), that's why I used >= 4 initially

Reply:
`assert len(rh_product.repository) == len(all_repo_ids) +1` would work :smile:

Reply:
Ah, so this test probably interferes with https://github.com/SatelliteQE/robottelo/blob/a9ac75db06365c20371d88381f89334f50347acf/tests/foreman/api/test_repositories.py#L148-L149
which uses `rhel9_bos` and `module_sca_manifest_org` too.

Anyway the `len(all_repo_ids) +1` would work until someone adds a test syncing different repo from the same product anywhere before this test. Now I feel torn apart what is better option. Keep the interference and `>= 4`, or use `function_sca_manifest_org` (which creates yet another allocation and downloads yet another manifest) and `== 4`.

Reply:
Well, since this test case is related to repositories bulk sync and not the repo-to-product relation, I'm fine with the `>=` assertion, @vijaysawant.